crates/core/transaction/src/plan.rs

```diff
@@ -68,18 +68,10 @@ impl TransactionPlan {
 
         let parameters_hash = self.transaction_parameters.effect_hash();
 
-        let memo_hash = self
-            .memo
-            .as_ref()
-            .map(|memo_plan| {
-                memo_plan
-                    .memo()
-                    .expect("can compute memo ciphertext")
-                    .effect_hash()
-            })
-            // If the memo is not present, use the all-zero hash to record its absence in
-            // the overall effect hash.
-            .unwrap_or_default();
+        let memo_hash = match self.memo {
+            Some(ref memo) => memo.memo()?.effect_hash(),
+            None => EffectHash::default(),
+        };
```

Comment:
this is a significant change in style compared to the previous implementation, not sure if that's appropriate

Reply:
I like the result of the change better, it's more succinct this way.